### ISSUE TITLE
[Enhancement] Mounting .chd using fuse fs

### ISSUE THREAD

**Issue by DanTheMan827** [mamedev/mame#4758]
It would be nice to be able to mount a .chd image and be able to access the individual components of the image.

For an optical disc dump, it could expose `.bin` and `.cue` files

just a suggestion since not every emulator supports .chd (which is a shame really)

**Comment by rb6502** [comment]
This is a great idea, but it would be an external project.  We already provide all of the necessary components under a very permissive BSD 2-clause license.

**Comment by RinMaru** [comment]
any update on this its a really needed feature

**Comment by rb6502** [comment]
Nobody has created such a project.  We don't have the bandwidth, but we can advise someone who wants to.

**Comment by cracyc** [comment]
FWIW, fuse isn't right for this because chd are block devices that can have a variety of filesystems.  [qemu-nbd](https://github.com/qemu/qemu/blob/master/qemu-nbd.c) would be a better starting point.

**Comment by cuavas** [comment]
Yeah, you essentially want something like the loopback device on Linux or the HDI driver on macOS that redirects a block device to the CHD file and allows the OS to mount the filesystem.  As far as I know, we don’t have anyone on the team with the necessary skills to do this.

**Comment by eadmaster** [comment]
i've made [this feature request to cdemu](https://sourceforge.net/p/cdemu/feature-requests/24/) some years ago, but no progress was made.

**Comment by ksv1986** [comment]
I've hacked together simple server for chds. No support for cd images for now, need to check if it even possible with nbd protocol.

https://github.com/ksv1986/rust-chdnbd

**Comment by lcipria** [comment]
@DanTheMan827 done

it's buggy and dumb as shit but "it works"

https://github.com/lcipria/py_chd_fuse

maybe you don't need it anymore, maybe yes? Please enjoy.
